pkg/autoscaler/metrics/collector.go

```diff
@@ -265,8 +294,8 @@ func newCollection(metric *av1alpha1.Metric, scraper StatsScraper, tickFactory f
 		stopCh: make(chan struct{}),
 	}
 
-	logger = logger.Named("collector").With(
-		zap.String(logkey.Key, fmt.Sprintf("%s/%s", metric.Namespace, metric.Name)))
+	key := types.NamespacedName{Namespace: metric.Namespace, Name: metric.Name}
+	logger = logger.Named("collector").With(zap.String(logkey.Key, key.String()))
```

Comment:
I'd just go `WithValue`.

Reply:
We do the `zap.String` thing everywhere we use `logkey.Key` and a stringified `"namespace/name"`